### PR TITLE
Password reset fixes

### PR DESCRIPTION
**Pull request by qazs**
Check for empty email string.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: common/models/user.js [hunk 1]
```diff
@@ -476,7 +476,10 @@ module.exports = function(User) {
             }
           });
         } else {
-          cb();
+          err = new Error("User '" + options.email + "' does not exist");
```
Comment (ritch) [inline common/models/user.js]: The caller of this function should already have the email... Why put it in the error message?

**Comment by ritch** [comment]
I'd be for merging this if we removed the `email` from the error message. Otherwise looks good.

**Comment by ritch** [comment]
Actually on second thought I don't think we should make any of these changes. Its adding behavior that I'm not comfortable with being the default.

Eg. you could use the error message to determine whether or not an account exists. If you'd really like this behavior just implement your own remote method. You could create your own version of the reset password method. You can also emit your own reset password event.

Closing for now (if you want it re-opened just chime in here and we can discuss).

**Comment by qazs** [comment]
The problem is there's no error message when an account doesn't exist, the response is the same as when a password is successfully being reset.

**Pull request closed without merging**